src/libertem/io/dataset/hdf5.py

```diff
@@ -291,6 +349,133 @@ def _get_tiles_normal(self, tiling_scheme, tileshape_nd, dest_dtype="float32"):
                     scheme_idx=scheme_idx,
                 )
 
+    def _get_reshaped_tiles(self, tiling_scheme, tileshape_nd, dest_dtype="float32"):
+        subslices = self._get_subslices(
+                tiling_scheme=tiling_scheme,
+                tileshape_nd=tileshape_nd,
+            )
+        # reshape tileshape_nd with respect to native shape
+        tileshape_nd = np.unravel_index(
+            np.ravel_multi_index(tuple(i - 1 for i in tileshape_nd), tileshape_nd),
+            self.native_shape
+        )
+        tileshape_nd = tuple(i + 1 for i in tileshape_nd)
+        with self.reader.get_h5ds() as dataset:
+            for scheme_idx, tile_slice in subslices:
+                sig_dims = self.native_shape.sig.dims
+                ds_shape_nav = self.meta.shape[:-sig_dims]
+                native_ds_shape_nav = self.native_shape[:-sig_dims]
+                o_nav = tile_slice.origin[:-sig_dims]
+                o_pos = np.ravel_multi_index(o_nav, ds_shape_nav)
+                s_nav = tile_slice.shape[:-sig_dims]
+                s_nav = tuple(a - 1 for a in s_nav)
+                s_sig = tile_slice.shape[-sig_dims:]
+                axis_size = np.size(dataset, len(native_ds_shape_nav) - 1)
+                s_pos_actual = np.ravel_multi_index(s_nav, ds_shape_nav)
+                frame_pos = o_pos + s_pos_actual
+                # if frame lies outside the image_count
+                if frame_pos >= self.meta.image_count:
+                    # skip tile if there are no frames in the slice
+                    if o_pos >= self.meta.image_count:
+                        break
+                    # last tile_slice, correct the shape according to number of frames in the slice
+                    else:
+                        s = tuple(a + 1 for a in np.unravel_index(
+                            self.meta.image_count - o_pos - 1, ds_shape_nav)
+                        ) + tuple(s_sig)
+                        tile_slice = Slice(
+                            shape=(Shape(s, sig_dims=sig_dims)), origin=tile_slice.origin
+                        )
+                # actual reshaping of the tile_slice
+                tile_slice = tile_slice.reshape(self.meta.shape, self.native_shape)
+                # check if tile_slice needs to split, then yield the slice/slices
+                yield from self._split_and_get_tile_slice(
+                    tile_slice, axis_size, tileshape_nd, scheme_idx
+                )
+
+    def _split_and_get_tile_slice(self, tile_slice, axis_size, tileshape_nd, scheme_idx):
```

Comment:
Could you add docstrings for the new methods? Especially a description of all parameters and how the recursion works (why it's guaranteed to arrive at the base case, for example) would be very helpful.